### PR TITLE
Add ability to specify which network error codes to retry on

### LINKED ISSUE #654: Option to not retry on timeout errors
I'm configuring requests for AWS lambda environment, and case is that:
- I set timeout option to ensure it ends before allowed lambda invocation time ends (so I handle eventual timeout gently by rejecting lambda invocation with meaningful error), and automated retry after such timeout occurs is unwanted.
- I'd like to have retries applied for case of temporary network or server unavailability errors.

Is there an option to achieve that?

### PR DESCRIPTION
**Pull request by szmarczak**
Fixes #654

### CODE REVIEW

File: test/retry.js [hunk 1]
```diff
@@ -151,6 +152,37 @@ test('custom retries', async t => {
 	t.true(tried);
 });
 
+test.only('custom errors', async t => {
```
Comment (brandon93s) [inline test/retry.js]: Remove `.only`

File: advanced-creation.md [hunk 2]
```diff
@@ -82,6 +82,16 @@ const defaults = {
 				502,
 				503,
 				504
+			],
+			errors: [
```
Comment (sindresorhus) [inline advanced-creation.md]: I think we should name it `errorCodes`, since they're from `error.code`.

Comment (sindresorhus) [reply advanced-creation.md]: Same in the other places you use "error".

Comment (szmarczak) [reply advanced-creation.md]: I agree.

**Pull request merged**